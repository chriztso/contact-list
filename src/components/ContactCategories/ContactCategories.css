.contact-categories{
    border: 1px solid rgb(233, 234, 240);
    display: flex; 
    padding: 5px 0px;     
    align-items: center;
    color: rgb(96, 102, 124);
    font-weight: 600;
}

.contact-categories__checkbox{
    margin-left: 12px;
    margin-right: 12px;
}

.contact-categories__contact-title{
    /* border: 1px solid white; */
    width: 20%; 
}

.contact-categories__value-title{
    /* border: 1px solid white; */
    width: 12%; 
    display: flex;
    align-items: center;
}

.contact-categories__location-title{
    /* border: 1px solid white; */
    width: 15%; 
}

.contact-categories__deals-title{
    /* border: 1px solid white; */
    width: 10%; 
}

.contact-categories__tags-title{
    /* border: 1px solid black; */
    width: 35%; 
}