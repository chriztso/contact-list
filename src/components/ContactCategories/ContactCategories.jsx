import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';
import './ContactCategories.css'

const ContactCategories = () => {
    return (
        <div className="contact-categories">
            <input type="checkbox" className="contact-categories__checkbox"></input>
            <span className="contact-categories__contact-title">Contact</span>
            <span className="contact-categories__value-title">
                Total Value
                <ArrowDropDownIcon />
            </span>
            <span className="contact-categories__location-title">Location</span>
            <span className="contact-categories__deals-title">Deals</span>
            <span className="contact-categories__tags-title">Tags</span>
            <MoreHorizIcon />
        </div>
    )
}

export default ContactCategories