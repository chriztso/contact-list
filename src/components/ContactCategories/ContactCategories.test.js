import { shallow } from 'enzyme';
import ContactCategories from '../ContactCategories/ContactCategories'
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';

describe('ContactCategories', () => {
    it('renders ContactCategories', () => {
        const wrapper = shallow(<ContactCategories />);
        expect(wrapper).toMatchSnapshot();
    })

    it('renders category names', () => {
       const wrapper = shallow(<ContactCategories />);
       expect(wrapper.find('.contact-categories__contact-title').text()).toEqual('Contact');
       expect(wrapper.find('.contact-categories__value-title').text()).toEqual('Total Value');
       expect(wrapper.find('.contact-categories__location-title').text()).toEqual('Location');
       expect(wrapper.find('.contact-categories__deals-title').text()).toEqual('Deals');
       expect(wrapper.find('.contact-categories__tags-title').text()).toEqual('Tags');
    })

    it('renders arrow down icon', () => {
       const wrapper = shallow(<ContactCategories />);
       expect(wrapper.find(ArrowDropDownIcon).length).toEqual(1);
    })

    it('renders three dots icon', () => {
        const wrapper = shallow(<ContactCategories />);
        expect(wrapper.find(MoreHorizIcon).length).toEqual(1);
     })
})