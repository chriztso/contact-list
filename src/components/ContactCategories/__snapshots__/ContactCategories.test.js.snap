// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactCategories renders ContactCategories 1`] = `
<div
  className="contact-categories"
>
  <input
    className="contact-categories__checkbox"
    type="checkbox"
  />
  <span
    className="contact-categories__contact-title"
  >
    Contact
  </span>
  <span
    className="contact-categories__value-title"
  >
    Total Value
    <Memo(ForwardRef(ArrowDropDownIcon)) />
  </span>
  <span
    className="contact-categories__location-title"
  >
    Location
  </span>
  <span
    className="contact-categories__deals-title"
  >
    Deals
  </span>
  <span
    className="contact-categories__tags-title"
  >
    Tags
  </span>
  <span
    className="contact-categories__three-dots"
  >
    <Memo(ForwardRef(MoreHorizIcon)) />
  </span>
</div>
`;
